internal/manager/run.go

```diff
@@ -312,3 +319,61 @@ func readyzHandler(mgr manager.Manager, dataplaneSynchronizer IsReady) func(*htt
 		return nil
 	}
 }
+
+type healthCheckHandler struct {
```

Comment:
I believe this deserves its own file.

Reply:
moved to another file and added unit tests.